### ISSUE TITLE
finetuning: although EVAL metrics are better, the script does not save the model checkpoint as “best”.

### ISSUE THREAD

**Issue by Mixomo** [erew123/alltalk_tts#359]
Since I started training XTTS, it always happened to me that even if the evaluation metrics in certain epochs are better, the script does not save the model as “best”.
It seems that it only saves the “best” in the first 10 epochs (the default value of epochs), and if I want to increase the number of training epochs, it happens what I commented above.
(Or otherwise, the model never seems to improve beyond the first 10 epochs, no matter how good the dataset is).

![image](https://github.com/user-attachments/assets/ad2e4b4d-aa18-4806-9e29-e324839c5f7e)

I also don't understand why in the LOSS graph even though it shows a decay, the AVG LOSS goes up.... 

![image](https://github.com/user-attachments/assets/05ab556a-ce4b-4fec-9a9c-a0b8fc1a2d90)

**Comment by Mixomo** [comment]
@erew123 Thank you for your quick response! No rush! :)
